src/clp_ffi_js/ir/StreamReader.hpp

```diff
@@ -123,6 +125,17 @@ class StreamReader {
      */
     [[nodiscard]] virtual auto decode_range(size_t begin_idx, size_t end_idx, bool use_filter) const
             -> DecodedResultsTsType = 0;
+    /**
+     * Finds the log event with the timestamp that's nearest to the `target_ts`.
+     * @param target_ts
+     * @return The index of the log event with:
+     * - the largest timestamp less than or equal to `target_ts`,
+     * - or the index `0` if all timestamps are greater than `target_ts`.
```

Comment:
```suggestion
     * Finds the log event, L, where if we assume:
     *
     * - the collection of log events is sorted in ascending timestamp order;
     * - and we insert a marker log event, M, with timestamp `target_ts` into the collection (if log
     *   events with timestamp `target_ts` already exist in the collection, M should be inserted
     *   after them).
     *
     * L is the event just before M, if M is not the first event in the collection; otherwise L is
     * the event just after M.
     *
     * @param target_ts
     * @return The index of the log event L.
```

I think this method is sufficiently unintuitive that we need to have a more specific description.